### PR TITLE
8341006: Optimize StackMapGenerator detect frames

### PR DESCRIPTION
**Pull request by wenshao**
1. Construct Frames directly without BitSet
2. Use Frame[] instead of ArrayList
3. Use EMPTY_FRAME_ARRAY for initialization. No need to allocate objects when there is no frame.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341006](https://bugs.openjdk.org/browse/JDK-8341006): Optimize StackMapGenerator detect frames (**Sub-task** - P4)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21183/head:pull/21183` \
`$ git checkout pull/21183`

Update a local copy of the PR: \
`$ git checkout pull/21183` \
`$ git pull https://git.openjdk.org/jdk.git pull/21183/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21183`

View PR using the GUI difftool: \
`$ git pr show -t 21183`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21183.diff">https://git.openjdk.org/jdk/pull/21183.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21183#issuecomment-2376524066)

### CODE REVIEW

**Review by liach — CHANGES_REQUESTED** [review]
The removal of bitset makes this frame detection significantly slower for startup. A heap sort or manual unrolling of bitset may be better.

Edit: the slowdown seems to be noise. I will investigate a bit more before leaving another review.

**Review by liach — APPROVED** [review]
This should be startup neutral.

File: src/java.base/share/classes/jdk/internal/classfile/impl/StackMapGenerator.java [hunk 1]
```diff
@@ -840,41 +848,33 @@ private IllegalArgumentException generatorError(String msg, int offset) {
     }
 
     /**
-     * Performs detection of mandatory stack map frames offsets
-     * in a single bytecode traversing pass
-     * @return <code>java.lang.BitSet</code> of detected frames offsets
+     * Performs detection of mandatory stack map frames in a single bytecode traversing pass
+     * @return detected frames
```
Comment (ExE-Boss) [inline src/java.base/share/classes/jdk/internal/classfile/impl/StackMapGenerator.java:852]: `@return` on `void` method:
```suggestion
```

**Pull request closed without merging**